tests/multigrid-global-coarsening/mg_transfer_p_06.cc

```diff
@@ -0,0 +1,150 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 - 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+/**
+ * Test transfer operator for polynomial coarsening where, in addition,
+ * FE_Nothing on the fine level is replaced by a valid coarse element
+ * on the coarse level.
+ *
+ * Example:
+ *
+ * +--+--+      +--+--+
+ * |kf|kf|      |kc|X |
+ * |--+--|  ->  |--+--|
+ * |kf|kf|      |kc|X |
+ * +--+--+      +--+--+
+ *
+ *   ... with fe_degree in the cells
+ *   ... with X being FE_Nothing.
+ */
+
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/mpi.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_tools.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+
+#include <deal.II/hp/fe_collection.h>
+
+#include <deal.II/matrix_free/fe_evaluation.h>
+#include <deal.II/matrix_free/matrix_free.h>
+
+#include <deal.II/multigrid/mg_transfer_global_coarsening.h>
+
+#include "mg_transfer_util.h"
+
+using namespace dealii;
+
+template <int dim, typename Number>
+void
+do_test(const FiniteElement<dim> &fe_fine, const FiniteElement<dim> &fe_coarse)
+{
+  parallel::distributed::Triangulation<dim> tria(
+    MPI_COMM_WORLD,
+    Triangulation<dim>::MeshSmoothing::none,
+    parallel::distributed::Triangulation<dim>::construct_multigrid_hierarchy);
```

Comment:
Do we need to run this test case with MPI, and if yes, do we need to run with both 1 and 4 processes? I think the bug is already exposed in serial, so I would suggest to merely run with a `dealii::Triangulation`. This avoids dependence on p4est and thus gets more widely tested. If you see value in testing the parallel infrastructure, I would suggest only running the code with 4 processes (or 3), and use the 1 process only if there are algorithmic differences between the two that should be captured.

Reply:
Sorry for the misunderstanding. @kronbichler's suggestion makes sense.

Reply:
`MGTwoLevelTransfer` does not seem to work without MPI. I now removed the test with 1 process.